agent/consul/txn_endpoint_test.go

```diff
@@ -554,57 +554,64 @@ func TestTxn_Apply_ACLDeny(t *testing.T) {
 	}
 
 	// Verify the transaction's return value.
-	var expected structs.TxnResponse
+	var outPos int
```

Comment:
Can you explain the intent here please?

Reply:
The broad purpose here is to perform a slightly stricter test on what we return; in particular make sure we have the right resource and permission information. 

The original code attempted to build an identical result to the expected return value from the API call and directly compare them, but that seemed impractical given the new structure. So I altered this into an element by element comparison inline..